src/tests/test_circumcision.py

```diff
@@ -0,0 +1,345 @@
+
+    # evolve population
+    while pop.date <= stop_date:
+        # advance ages and birth circumcision
+        pop.data.age += time_step.days / 365
+        pop.circumcision.update_birth_circumcision(pop.data, time_step, pop.date)
+        pop.date += time_step
+
+    # get stats
+    no_male = sum((pop.data[col.SEX] == SexType.Male)
+                  & (pop.data[col.AGE] > 0.25))
+    no_circumcised, mean, stdev = get_birth_circ_stats(pop, no_male)
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_circumcised, pop.data)
+    # check that no circumcised people have undergone VMMC
+    assert sum(pop.data[col.CIRCUMCISED] & pop.data[col.VMMC]) == 0
+
+
+def test_vmmc_case_0():
+
+    test_ages = [18, 25, 40]
+    for age in test_ages:
+
+        N = 100000
+        start_date = date(2007, 1, 1)
+        stop_date = date(2010, 1, 1)
+        time_step = timedelta(days=90)
+
+        # build artificial population
+        pop = Population(size=N, start_date=start_date)
+        pop.data[col.SEX] = SexType.Male
+        pop.data[col.AGE] = age
+        reset_pop_circ(pop)
+
+        pop.circumcision.circ_policy_scenario = 0
+        set_covid(pop.circumcision, False)
+        set_vmmc_default_dates(pop.circumcision)
+
+        # evolve population for a year
+        for i in range(0, 5):
+            pop.data.age += time_step.days / 365
+            pop.circumcision.update_vmmc(pop)
+            pop.date += time_step
+        # check no VMMC occurs until vmmc_start_year
+        assert sum(pop.data[col.VMMC]) == 0
+
+        # evolve population during vmmc_start_year
+        pop.data.age += time_step.days / 365
+        pop.circumcision.update_vmmc(pop)
+
+        # get stats
+        prob_circ = pop.circumcision.calc_prob_circ(test_ages.index(age)+1)
+        no_vmmc, mean, stdev = get_vmmc_stats(pop, prob_circ)
+        # basic checks
+        general_circumcision_checks(mean, stdev, no_vmmc, pop.data)
+        # nobody over 50 has been circumcised
+        assert sum((pop.data[col.AGE] >= 50) & (pop.data[col.VMMC])) == 0
+
+        # evolve population for a few more years
+        while pop.date <= stop_date:
+            circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male)
+                                        & pop.data[col.CIRCUMCISED]]
+            # advance ages and vmmc
+            pop.data.age += time_step.days / 365
+            pop.circumcision.update_vmmc(pop)
+            pop.date += time_step
+            # check circumcisied people remain circumcised each step
+            new_circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male)
+                                            & pop.data[col.CIRCUMCISED]]
+            assert circ_males.isin(new_circ_males).all()
+
+        # nobody under 10 has been circumcised
+        assert sum((pop.data[col.AGE] < 10) & (pop.data[col.VMMC])) == 0
+
+
+def test_vmmc_case_1():
+
+    test_ages = [18, 25, 40]
```

Comment:
Should this include younger ages so we can see the effect on 10-14 year olds?

Reply:
Seems reasonable, I can add 14 as a test age where we expect to see no VMMC after the policy intervention year.